### ISSUE TITLE
IDISP025 should not report abstract methods

### ISSUE THREAD

**Issue by sunsided** [DotNetAnalyzers/IDisposableAnalyzers#226]
The following method is triggering the [IDISP025](https://github.com/DotNetAnalyzers/IDisposableAnalyzers/blob/master/documentation/IDISP025.md) warning "Class with no virtual dispose method should be sealed":

```csharp
public abstract void Dispose();
```

Since it is effectively the same as

```csharp
public virtual void Dispose {}
```

plus some guarantees that the method will definitely be implemented, it shouldn't raise the warning.

**Comment by JohanLarsson** [comment]
Dumbed that one, thanks for reporting.

**Comment by sunsided** [comment]
Thanks for this project! :bow:

**Comment by sunsided** [comment]
@JohanLarsson I just noticed that a similar issue can be triggered with the `Dispose(bool disposing)` variant:

```csharp
protected virtual void Dispose(bool disposing) 
{
}
```

works as expected, but

```
protected abstract void Dispose(bool disposing);
```

triggers an `IDISP025` warning.
